test/gui/shared/scripts/bdd_hooks.py

```diff
@@ -49,6 +50,18 @@ def hook(context):
 
     if context.userData['clientConfigFile'] == '':
         context.userData['clientConfigFile']='-'
+        
+    req = urllib.request.Request(
+        context.userData['middlewareUrl'] + 'init',
```

Comment:
can you use function to join paths such as `path.join()` here. it will throw error if middlewareUrl doesn't end with `/`